webui/react/src/components/NavigationTabbar.tsx

```diff
@@ -118,13 +127,30 @@ const NavigationTabbar: React.FC = () => {
             path: paths.workspaceList(),
           },
           ...Loadable.match(pinnedWorkspaces, {
-            Loaded: (workspaces) =>
-              workspaces.map((workspace) => ({
+            Loaded: (workspaces) => {
+              const workspaceIcons = workspaces.map((workspace) => ({
                 icon: <DynamicIcon name={workspace.name} size={24} style={{ color: 'black' }} />,
                 label: workspace.name,
                 onClick: (e: AnyMouseEvent) =>
                   handlePathUpdate(e, paths.workspaceDetails(workspace.id)),
-              })),
+              }));
+              if (canCreateWorkspace) {
+                return workspaceIcons.concat([
```

Comment:
nit: we return `workspaceIcons` anyways, so i think using `push` is better instead of return early using `concat` for readability
also, `push` is fatster than `concat` in general 
https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki

Reply:
✅